homeassistant/components/sinch/config_flow.py

```diff
@@ -0,0 +1,205 @@
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+from homeassistant.helpers.typing import ConfigType
+
+from .const import (
+    CONF_DEFAULT_RECIPIENTS,
+    CONF_SERVICE_PLAN_ID,
+    DEFAULT_NAME,
+    DEFAULT_SENDER,
+    DOMAIN,
+    MAX_SENDER_LENGTH,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def check_client_connection(service_plan_id: str, api_key: str):
+    """Check client connection."""
+    client = Client(service_plan_id, api_key)
+    batches = client.fetch_batches()
+    batches.get(0)
+    return True
+
+
+class SinchFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for sinch."""
+
+    VERSION = 1
+    MINOR_VERSION = 1
+
+    async def async_step_user(
+        self, user_input: ConfigType | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is not None:
+            await self.async_set_unique_id(user_input[CONF_SERVICE_PLAN_ID])
+            self._abort_if_unique_id_configured()
+
+            self._async_abort_entries_match(
+                {
+                    CONF_NAME: user_input[CONF_NAME],
+                }
+            )
+            if not (
+                error := await self._async_try_connect(
+                    user_input[CONF_SERVICE_PLAN_ID], user_input[CONF_API_KEY]
+                )
+            ):
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME],
+                    data=user_input,
+                )
+            errors["base"] = error
+
+        else:
+            user_input = {
+                CONF_NAME: DEFAULT_NAME,
+                CONF_SERVICE_PLAN_ID: "",
+                CONF_API_KEY: "",
+            }
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_NAME, default=user_input[CONF_NAME]): str,
+                    vol.Required(
+                        CONF_SERVICE_PLAN_ID, default=user_input[CONF_SERVICE_PLAN_ID]
+                    ): str,
+                    vol.Required(CONF_API_KEY, default=user_input[CONF_API_KEY]): str,
+                }
+            ),
+            errors=errors,
+            description_placeholders={
+                "token_url": "https://dashboard.sinch.com/sms/api/services",
+                "component_url": ("https://www.home-assistant.io/integrations/sinch/"),
+            },
+        )
+
+    async def async_step_import(self, import_config: ConfigType) -> ConfigFlowResult:
+        """Import a config entry from configuration.yaml."""
+        _LOGGER.warning(
+            "Configuration of the Sinch integration in YAML is deprecated and "
+            "will be removed in a future release; Your existing configuration "
+            "has been imported into the UI automatically and can be safely removed "
+            "from your configuration.yaml file"
+        )
```

Comment:
Raise an issue, for example as done here: https://github.com/home-assistant/core/blob/4ff64534d37dfd472a23ad3ee58d074ee41bf93e/homeassistant/components/downloader/__init__.py#L85-L98